File: src/types/property.ts
import type { HazardType, RiskSeverity } from './risk';

export interface FloodZoneDetail {
  in_arpsi_zone: boolean;
  zone_id: string | null;
  zone_name: string | null;
  zone_type: string | null;
  return_period: string | null;
  risk_level: string | null;
  distance_to_nearest_zone_m: number | null;
}

export interface WildfireProximityDetail {
  elevation_m: number;
  slope_pct: number;
  modifier: number;
  explanation: string;
}

export interface TerrainDetail {
  elevation_m: number;
  slope_pct: number;
}

export interface HazardScoreDetail {
  score: number;
  severity: RiskSeverity;
  province_score: number;
  modifier: number;
  explanation: string;
}

export interface PropertyReportResponse {
  report_id: string;
  address_text: string;
  formatted_address: string;
  latitude: number;
  longitude: number;
  province_code: string;
  province_name: string;
  municipality_code: string | null;

  composite_score: number;
  dominant_hazard: HazardType | string;
  severity: RiskSeverity;

  flood: HazardScoreDetail;
  wildfire: HazardScoreDetail;
  heatwave: HazardScoreDetail;
  drought: HazardScoreDetail;
  coldwave: HazardScoreDetail;
  windstorm: HazardScoreDetail;
  seismic: HazardScoreDetail;

  flood_zone: FloodZoneDetail;
  wildfire_proximity: WildfireProximityDetail;
  terrain: TerrainDetail;

  computed_at: string;
  expires_at: string | null;
  pdf_available: boolean;
  share_url: string;
}

export interface PropertyReportListItem {
  report_id: string;
  address_text: string;
  formatted_address: string;
  composite_score: number;
  severity: RiskSeverity;
  dominant_hazard: string;
  computed_at: string;
}

File: backend/app/schemas/property.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class PropertyReportRequest(BaseModel):
    address: str = Field(max_length=500)
    include_pdf: bool = False


class GeocodingResponse(BaseModel):
    formatted_address: str
    latitude: float
    longitude: float
    province_code: str
    municipality_code: str | None = None
    confidence: float


class FloodZoneDetail(BaseModel):
    in_arpsi_zone: bool
    zone_id: str | None = None
    zone_name: str | None = None
    zone_type: str | None = None  # fluvial, pluvial, coastal
    return_period: str | None = None  # T10, T100, T500
    risk_level: str | None = None
    distance_to_nearest_zone_m: float | None = None


class WildfireProximityDetail(BaseModel):
    elevation_m: float
    slope_pct: float
    modifier: float
    explanation: str


class TerrainDetail(BaseModel):
    elevation_m: float
    slope_pct: float


class HazardScoreDetail(BaseModel):
    score: float
    severity: str  # low, moderate, high, very_high, critical
    province_score: float
    modifier: float
    explanation: str


class PropertyReportResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    report_id: str
    address_text: str
    formatted_address: str
    latitude: float
    longitude: float
    province_code: str
    province_name: str
    municipality_code: str | None = None

    composite_score: float
    dominant_hazard: str
    severity: str

    flood: HazardScoreDetail
    wildfire: HazardScoreDetail
    heatwave: HazardScoreDetail
    drought: HazardScoreDetail
    coldwave: HazardScoreDetail
    windstorm: HazardScoreDetail
    seismic: HazardScoreDetail

    flood_zone: FloodZoneDetail
    wildfire_proximity: WildfireProximityDetail
    terrain: TerrainDetail

    computed_at: str  # ISO datetime string
    expires_at: str | None = None
    pdf_available: bool = False
    share_url: str


class PropertyReportListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    report_id: str
    address_text: str
    formatted_address: str
    composite_score: float
    severity: str
    dominant_hazard: str
    computed_at: str
